packages/lodestar-validator/src/validator.ts

```diff
@@ -73,6 +75,54 @@ export class Validator {
     if (this.blockService) await this.blockService.stop();
   }
 
+  /**
+   * Perform a voluntary exit for the given validator by its public key.
+   */
+  public async voluntaryExit(publicKey: string): Promise<void> {
+    await this.apiClient.connect();
+
+    const stateValidator = await this.apiClient.beacon.state.getStateValidator(
+      "head",
+      this.config.types.BLSPubkey.fromJson(publicKey)
+    );
+    if (!stateValidator) throw new Error("No validator found in validator store.");
+
+    const epoch = this.apiClient.clock.currentEpoch;
+
+    const voluntaryExit = {
+      epoch,
+      validatorIndex: stateValidator.index,
+    };
+
+    const state: BeaconState = this.config.types.BeaconState.tree.defaultValue();
+    state.fork = (await this.apiClient.beacon.state.getFork("head")) || state.fork;
+    state.genesisValidatorsRoot =
+      (await this.apiClient.beacon.getGenesis())?.genesisValidatorsRoot || state.genesisValidatorsRoot;
+    const domain = getDomain(this.config, state, DomainType.VOLUNTARY_EXIT, epoch);
+    const signingRoot = computeSigningRoot(this.config, this.config.types.VoluntaryExit, voluntaryExit, domain);
+
+    let secretKey;
+    for (const sk of this.opts.secretKeys) {
+      if (sk.toPublicKey().toHex() === publicKey) secretKey = sk;
+    }
+    if (!secretKey) throw new Error(`No matching secret key found for public key ${publicKey}`);
+
+    const signedVoluntaryExit: SignedVoluntaryExit = {
+      message: voluntaryExit,
+      signature: secretKey.sign(signingRoot).toBytes(),
+    };
+
+    try {
+      this.logger.info(`Waiting for validator ${publicKey} to be exited...`);
+      await this.apiClient.beacon.pool.submitVoluntaryExit(signedVoluntaryExit);
+      this.logger.info(`Successfully exited validator ${publicKey}`);
```

Comment:
this is not exactly true, validator will first wait for exit to be included in block than wait in queue.

Maybe better is to write submitted voluntary exit to the network.